rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -574,15 +593,21 @@ void SPUThread::process_mfc_cmd()
 	auto check_queue_size = [&]()
 	{
 		while (mfc_queue.size() >= 16)
-		{
-			if (test(state, cpu_flag::stop + cpu_flag::dbg_global_stop))
-			{
-				return;
-			}
-
-			// TODO: investigate lost notifications
-			std::this_thread::yield();
-			_mm_lfence();
+ 		if (mfc_queue.size() >= 16)
+ 		{
+			do {
+ 				if (test(state, cpu_flag::stop + cpu_flag::dbg_global_stop))
+ 				{
+ 					return;
+ 				}
+ 				
+ 				// TODO: investigate lost notifications
+ 				std::this_thread::yield();
+ 				_mm_lfence();
+ 				
+ 			} while (mfc_queue.size() >= 16);
+ 		
+ 			ch_event_stat |= SPU_EVENT_QV;
```

Comment:
This seems like a weird place to put this. Depending on what this event is supposed to actually signal, what will probably normally happen is the queue is full, it drops down by 1, we signal the event, but still push the cmd, and its full again? i assume this check and event signal should be elsewhere